File: src/com/ptby/dynamicreturntypeplugin/config/ProjectDynamicReturnTypeMap.java
package com.ptby.dynamicreturntypeplugin.config;

import com.intellij.openapi.project.Project;

import java.util.concurrent.ConcurrentHashMap;

public class ProjectDynamicReturnTypeMap extends ConcurrentHashMap<String, DynamicReturnTypeConfig> {

    public DynamicReturnTypeConfig get( Project project ) {

        String  key = project.getBaseDir().toString();
        DynamicReturnTypeConfig dynamicReturnTypeConfig = get( key );
        if ( dynamicReturnTypeConfig == null ) {
            return new DynamicReturnTypeConfig();
        }

        return dynamicReturnTypeConfig;
    }


    public void put( Project project, DynamicReturnTypeConfig dynamicReturnTypeConfig ) {
        String  key = project.getBaseDir().toString();
        put( key, dynamicReturnTypeConfig );
    }


    public void resetProject( Project project  ) {
        String key = project.getBaseDir().toString();
        put( key, new DynamicReturnTypeConfig() );
    }
}

File: src/com/ptby/dynamicreturntypeplugin/config/ProjectDynamicReturnTypeMap.kt
package com.ptby.dynamicreturntypeplugin.config

import com.intellij.openapi.project.Project

import java.util.concurrent.ConcurrentHashMap

public class ProjectDynamicReturnTypeMap : ConcurrentHashMap<String, DynamicReturnTypeConfig>() {

    fun get(project: Project): DynamicReturnTypeConfig {
        val key = project.getBaseDir().toString()
        val dynamicReturnTypeConfig = get(key)
                ?: return DynamicReturnTypeConfig.newEmpty()

        return dynamicReturnTypeConfig
    }


    fun put(project: Project, dynamicReturnTypeConfig: DynamicReturnTypeConfig) {
        val key = project.getBaseDir().toString()
        put(key, dynamicReturnTypeConfig)
    }


    fun resetProject(project: Project) {
        val key = project.getBaseDir().toString()
        put(key, DynamicReturnTypeConfig.newEmpty())
    }
}
